homeassistant/components/aemet/sensor.py

```diff
@@ -232,108 +354,61 @@ async def async_setup_entry(
 ) -> None:
     """Set up AEMET OpenData sensor entities based on a config entry."""
     domain_data = hass.data[DOMAIN][config_entry.entry_id]
-    name = domain_data[ENTRY_NAME]
-    weather_coordinator = domain_data[ENTRY_WEATHER_COORDINATOR]
+    name: str = domain_data[ENTRY_NAME]
+    coordinator: WeatherUpdateCoordinator = domain_data[ENTRY_WEATHER_COORDINATOR]
 
-    unique_id = config_entry.unique_id
-    entities: list[AbstractAemetSensor] = [
-        AemetSensor(name, unique_id, weather_coordinator, description)
-        for description in WEATHER_SENSOR_TYPES
-        if description.key in MONITORED_CONDITIONS
-    ]
-    entities.extend(
-        [
-            AemetForecastSensor(
-                f"{domain_data[ENTRY_NAME]} {mode} Forecast",
-                f"{unique_id}-forecast-{mode}",
-                weather_coordinator,
-                mode,
-                description,
+    entities: list[AemetSensor] = []
+
+    for description in FORECAST_SENSORS:
```

Comment:
```suggestion
    for description in FORECAST_SENSORS + WEATHER_SENSORS:
```

Reply:
Done in fdfbf312a22b98f0eb60e2190e6e5616031c7635